Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift

```diff
@@ -91,43 +122,46 @@ public struct ApolloCodegenConfiguration {
       /// is expecting the generated files in the configured output location.
       case other
     }
-
-    /// Local path where the generated schema types files should be stored.
-    public let path: String
-    /// Automation to ease the integration of the generated schema types file with compatible dependency managers.
-    public let moduleType: ModuleType
-    /// Name used to scope the generated schema type files.
-    public let schemaName: String
-
-    /// Designated initializer.
-    ///
-    /// - Parameters:
-    ///  - path: Local path where the generated schema type files should be stored.
-    ///  - schemaName: Name used to scope the generated schema type files.
-    ///  - moduleType: Type of module that will be created for the schema types files. Defaults to `.none`.
-    public init(
-      path: String,
-      schemaName: String,
-      moduleType: ModuleType = .none
-    ) {
-      self.path = path
-      self.schemaName = schemaName
-      self.moduleType = moduleType
-    }
   }
 
   /// The local path structure for the generated operation object files.
   public enum OperationsFileOutput: Equatable {
     /// All operation object files will be located in the module with the schema types.
     case inSchemaModule
-    /// Operation object files will be co-located relative to the defining operation `.graphql` file. If `subpath` is specified a subfolder
-    /// will be created relative to the `.graphql` file and the operation object files will be generated there. If no `subpath` is
-    /// defined then all operation object files will be generated alongside the `.graphql` file.
+    /// Operation object files will be co-located relative to the defining operation `.graphql`
+    /// file. If `subpath` is specified a subfolder will be created relative to the `.graphql` file
+    /// and the operation object files will be generated there. If no `subpath` is defined then all
+    /// operation object files will be generated alongside the `.graphql` file.
     case relative(subpath: String?)
     /// All operation object files will be located in the specified path.
     case absolute(path: String)
   }
 
+  /// The local path structure for the generated test mock object files.
+  public enum TestMockFileOutput: Equatable {
```

Comment:
@calvincestari Let me know what you think about this approach. After a lot more consideration, this is what I felt was the simplest and cleanest way to provide all of the configuration we wanted with the least complexity added to the configuration structure.

Reply:
It should satisfy what we're wanting to allow for. As more complicated use cases come up we can reconsider.